src/lib/data/source-text.js

```diff
@@ -34,6 +34,8 @@ export default class SourceText {
     } else {
       this.metadata = new Metadata()
     }
+
+    this.detectedLang = false

```

Comment:
this is a flag for checking if language was already detected, to avoid double checks (to safe API requests)